homeassistant/components/mikrotik/hub.py

```diff
@@ -354,21 +358,9 @@ async def async_add_options(self):
                 self.config_entry, data=data, options=options
             )
 
-    async def request_update(self):
-        """Request an update."""
-        if self.progress is not None:
-            await self.progress
-            return
-
-        self.progress = self.hass.async_create_task(self.async_update())
-        await self.progress
-
-        self.progress = None
-
     async def async_update(self):
         """Update Mikrotik devices information."""
-        await self.hass.async_add_executor_job(self._mk_data.update)
-        async_dispatcher_send(self.hass, self.signal_update)
+        await self.hass.async_add_executor_job(self._mk_data.update_devices)
 
     async def async_setup(self):
```

Comment:
It is a bit confusing to have this so far away from async_setup_entry and bound to a coordinator object.

It would be better to move it into async_setup_entry in `__init__` to be in-line with the rest of the codebase

Also change `self.async_refresh()` to `coordinator.async_config_entry_first_refresh()`

Reply:
I updated the code and had to update the tests also to pass. Appreciate your review on these changes